src/zjs_ipm.h

```diff
@@ -95,6 +124,15 @@ typedef struct zjs_ipm_message {
             uint8_t color_b;
             void *buffer;
         } glcd;
+#endif
+
+#ifdef BUILD_MODULE_SENSOR
+        struct sensor_data {
+            enum sensor_channel channel;
+            uint32_t frequency;
+            double value[3];
```

Comment:
@riju I updated patch to use a union to hold the reading values depending on the sensor type